axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/ProjectBusinessServiceImpl.java

```diff
@@ -49,7 +63,81 @@ public ProjectBusinessServiceImpl(ProjectRepository projectRepository) {
     SaleOrder order = Beans.get(SaleOrderCreateService.class).createSaleOrder(project.getCompany());
-    order.setClientPartner(project.getClientPartner());
+
+    Partner clientPartner = project.getClientPartner();
+    Company company = project.getCompany();
+
+    order.setProject(projectRepo.find(project.getId()));
+    order.setClientPartner(clientPartner);
+    order.setCompany(company);
+
+    order.setMainInvoicingAddress(partnerService.getInvoicingAddress(clientPartner));
+    order.setMainInvoicingAddressStr(
+        addressService.computeAddressStr(order.getMainInvoicingAddress()));
+    order.setDeliveryAddress(partnerService.getDeliveryAddress(clientPartner));
+    order.setDeliveryAddressStr(addressService.computeAddressStr(order.getDeliveryAddress()));
+    order.setContactPartner(project.getContactPartner());
+    order.setIsNeedingConformityCertificate(clientPartner.getIsNeedingConformityCertificate());
+    order.setCompanyBankDetails(
+        Beans.get(AccountingSituationService.class)
+            .getCompanySalesBankDetails(company, clientPartner));
+
+    if (project.getCurrency() != null) {
+      order.setCurrency(project.getCurrency());
+    } else if (clientPartner.getCurrency() != null) {
+      order.setCurrency(clientPartner.getCurrency());
+    } else {
+      order.setCurrency(company.getCurrency());
+    }
+
+    if (project.getPriceList() != null) {
+      order.setPriceList(project.getPriceList());
+    } else {
+      order.setPriceList(
+          Beans.get(PartnerPriceListService.class)
+              .getDefaultPriceList(clientPartner, PriceListRepository.TYPE_SALE));
+    }
+
+    if (order.getPriceList() != null) {
+      order.setHideDiscount(order.getPriceList().getHideDiscount());
+    }
+
+    if (clientPartner.getPaymentCondition() != null) {
+      order.setPaymentCondition(clientPartner.getPaymentCondition());
+    } else {
+      order.setPaymentCondition(company.getAccountConfig().getDefPaymentCondition());
+    }
+
+    if (clientPartner.getInPaymentMode() != null) {
+      order.setPaymentMode(clientPartner.getInPaymentMode());
+    } else {
+      order.setPaymentMode(company.getAccountConfig().getInPaymentMode());
+    }
+
+    if (order.getDuration() != null && order.getCreationDate() != null) {
+      order.setEndOfValidityDate(
+          Beans.get(DurationService.class)
+              .computeDuration(order.getDuration(), order.getCreationDate()));
+    }
+
+    App app = Beans.get(AppBaseService.class).getApp("supplychain");
+    if (app != null) {
+      order.setShipmentMode(clientPartner.getShipmentMode());
+      order.setFreightCarrierMode(clientPartner.getFreightCarrierMode());
+      if (clientPartner.getFreightCarrierMode() != null) {
+        order.setCarrierPartner(clientPartner.getFreightCarrierMode().getCarrierPartner());
+      }
+      Boolean interco =
+          Beans.get(AppSupplychainService.class).getAppSupplychain().getIntercoFromSale()
```

Comment:
Use 'Beans.get(AppSupplychainService.class).getAppSupplychain().' before  'if' statement and remove 'App app'. 
Check appSupplyChain not nulll and active inside if condition.